cni/pkg/ambient/informers_test.go

```diff
@@ -184,7 +182,6 @@ func TestServerReconcilePod(t *testing.T) {
 
 		t.Run(c.name, func(t *testing.T) {
 			ctx, cancel := context.WithCancel(ctx)
-			defer cancel()
```

Comment:
if the move is not a must, please restore

Reply:
I call it manually in line 199 to make sure all the events handled by informer.

Reply:
cancel often comes along with context init. what is the intention to stop kubeclient before